api/migrations/versions/bda5cebe15fd_indabax_registration.py

```diff
@@ -0,0 +1,676 @@
+        self.section_id = section_id
+        self.headline = headline
+        self.placeholder = placeholder
+        self.order = order
+        self.type = type
+        self.description = description
+        self.options = options
+        self.is_required = is_required
+        self.validation_regex = validation_regex
+        self.validation_text = validation_text
+        self.required_value = required_value
+
+class Country(Base):
+    __tablename__ = 'country'
+    __table_args__ = {'extend_existing': True}
+
+    id = db.Column(db.Integer(), primary_key=True)
+    name = db.Column(db.String(100), nullable=False)
+
+    def __init__(self, name):
+        self.name = name
+
+class TagType(Enum):
+    RESPONSE = 'response'
+    REGISTRATION = 'registration'
+    GRANT = 'grant'
+
+class Tag(Base):
+    __tablename__ = 'tag'
+    __table_args__ = {'extend_existing': True}
+
+    id = db.Column(db.Integer(), primary_key=True)
+    event_id = db.Column(db.Integer(), db.ForeignKey('event.id'), nullable=False)
+    tag_type = db.Column(db.Enum(TagType), nullable=True)
+    active = db.Column(db.Boolean(), nullable=False, default=True)
+
+class TagTranslation(db.Model):
+    __tablename__ = 'tag_translation'
+    __table_args__ = (
+        db.UniqueConstraint('tag_id', 'language', name='uq_tag_id_language'),
+        {'extend_existing': True}
+    )
+    id = db.Column(db.Integer(), primary_key=True)
+    tag_id = db.Column(db.Integer(), db.ForeignKey('tag.id'), nullable=False)
+    language = db.Column(db.String(2), nullable=False)
+    name = db.Column(db.String(), nullable=False)
+    description = db.Column(db.String(), nullable=True)
+
+def get_country_list(session):
+    countries = session.query(Country).all()
+    country_list = []
+    for country in countries:
+        country_list.append({
+            'label': country.name,
+            'value': country.name
+        })
+    return country_list
+
+def upgrade():
+    Base.metadata.bind = op.get_bind()
+    session = orm.Session(bind=Base.metadata.bind)
+
+    event = session.query(Event).filter_by(key='indabax2023').first()
+    if not event:
+        org = session.query(Organisation).filter_by(domain='indabax').first()
+        event = Event(
+            None,
+            None,
+            datetime(2023, 7, 12),
+            datetime(2023, 7, 14),
+            'indabax2023',
+            org.id,
+            'info@indabax.co.za',
+            'https://indabax.co.za',
+            datetime(2023, 5, 22),
+            datetime(2023, 6, 12),
+            datetime(2023, 6, 10),
+            datetime(2023, 6, 18),
+            datetime(2023, 6, 15),
+            datetime(2023, 6, 20),
+            datetime(2023, 6, 16),
+            datetime(2023, 7, 23),
+            datetime(2023, 6, 16),
+            datetime(2023, 7, 12),
+            EventType.EVENT,
+            True
+        )
+        session.add(event)
+        session.commit()
```

Comment:
Flagging this bit since this isn't something that's reverted in the downgrade. To get the upgrade to run it required this event to already have been created (which was done outside of the migrations). Not sure how else to handle this to get the migration to pass.